# Peer-graded Assignment: Capstone Project Notebook
### Created by: Eric J. Puttock

# Assignment Week 1:

Import pandas and numpy libraries as pd and np, respectively.

In [1]:
import pandas as pd
import numpy as np

Print out "Hello Capstone Project Course!".

In [2]:
print('Hello Capstone Project Course!')

Hello Capstone Project Course!


# Assignment Week 2:

We will read in data from the table from the following Wikipedia page listing postal codes of Canada.

In [3]:
WikiLink = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'

The data that from the table is read into a pandas dataframe.

In [4]:
wikidf = pd.read_html(WikiLink)[0]
wikidf

Postal Code           Borough  \
0           M1A      Not assigned   
1           M2A      Not assigned   
2           M3A        North York   
3           M4A        North York   
4           M5A  Downtown Toronto   
..          ...               ...   
175         M5Z      Not assigned   
176         M6Z      Not assigned   
177         M7Z      Not assigned   
178         M8Z         Etobicoke   
179         M9Z      Not assigned   

                                         Neighbourhood  
0                                         Not assigned  
1                                         Not assigned  
2                                            Parkwoods  
3                                     Victoria Village  
4                            Regent Park, Harbourfront  
..                                                 ...  
175                                       Not assigned  
176                                       Not assigned  
177                                       Not assigned  
178  Mimico NW, The Queensway West, South of Bloor,...  
179                                       Not assigned  

[180 rows x 3 columns]

We remove the rows of 'Borough' with 'Not assigned' from the dataframe (and reset index).

In [5]:
wikidf = wikidf[wikidf['Borough'] != 'Not assigned'].reset_index(drop=True)
wikidf

Postal Code           Borough  \
0           M3A        North York   
1           M4A        North York   
2           M5A  Downtown Toronto   
3           M6A        North York   
4           M7A  Downtown Toronto   
..          ...               ...   
98          M8X         Etobicoke   
99          M4Y  Downtown Toronto   
100         M7Y      East Toronto   
101         M8Y         Etobicoke   
102         M8Z         Etobicoke   

                                         Neighbourhood  
0                                            Parkwoods  
1                                     Victoria Village  
2                            Regent Park, Harbourfront  
3                     Lawrence Manor, Lawrence Heights  
4          Queen's Park, Ontario Provincial Government  
..                                                 ...  
98       The Kingsway, Montgomery Road, Old Mill North  
99                                Church and Wellesley  
100  Business reply mail Processing Centre, South C...  
101  Old Mill South, King's Mill Park, Sunnylea, Hu...  
102  Mimico NW, The Queensway West, South of Bloor,...  

[103 rows x 3 columns]

If a cell has a 'Borough' but a 'Not assigned' 'Neighborhood', then the 'Neighborhood' should be assigned the same as borough.
However, as shown below, there aren't any 'Neighbourhood' rows with 'Not assigned' after removal of 'Not assigned' 'Borough.

Therefore, no additional effort will be made to code that portion.

In [6]:
wikidf[wikidf['Neighbourhood']=='Not assigned']

Empty DataFrame
Columns: [Postal Code, Borough, Neighbourhood]
Index: []

Here, we show our cleaned up dataframe contain 103 rows (Postal Codes) from Toronto.

In [7]:
wikidf.shape

(103, 3)